test-runner/src/main.ts

```diff
@@ -94,13 +96,33 @@ function parsePaths() {
   }
 }
 
+async function fetchApiBuildInfo(apiUrl: URL): Promise<BuildInfo | null> {
+  try {
+    apiUrl.pathname = '/version'
```

Comment:
We shouldn't directly modify the passed `apiUrl` here since it can cause unpredicted side-effects.
Let's either pass a clone of `apiUrl` here, or just a string from which we will construct new `URL` instance.